softioc/extension.c

```diff
@@ -20,14 +20,17 @@
 
 /* Helper for function below. */
 #define ADD_ENUM(dict, name) \
-    PyDict_SetItemString(dict, #name, PyInt_FromLong(name))
+    tmp = PyInt_FromLong(name); \
+    PyDict_SetItemString(dict, #name, tmp); \
+    Py_XDECREF(tmp);
```

Comment:
And do we need this here too?

Reply:
Ick.  Yes (and you can use `Py_DECREF` again).

To reduce the ick factor at this point I might just make a helper function
```c
static void add_enum(PyObject *dict, const char *name, long value)
{
    PyObject *py_value = PyInt_FromLong(value);
    PyDict_SetItemString(dict, name, py_value);
    Py_DECREF(py_value);
}
```
and either call that directly or simplify `ADD_ENUM` to
```c
#define ADD_ENUM(dict, name) add_enum(dict, #name, name)
```

An interesting thought on terminology.  The documentation for `PyDict_SetItemString` says
> This function _does not_ steal a reference to `val`.

I guess today I'd rather reword that as
> This function does not take ownership of `val`